VocaDb.ResXFileCodeGenerator/SourceGenerator.cs

```diff
@@ -1,146 +1,41 @@
-using System.Globalization;
 using Microsoft.CodeAnalysis;
 
 namespace VocaDb.ResXFileCodeGenerator;
 
 [Generator]
-public class SourceGenerator : ISourceGenerator
+public class SourceGenerator : IIncrementalGenerator
 {
 	private static readonly IGenerator s_generator = new StringBuilderGenerator();
 
-	// Code from: https://github.com/dotnet/ResXResourceManager/blob/0ec11bae232151400a5a8ca7b9835ac063c516d0/src/ResXManager.Model/ResourceManager.cs#L267
-	private static bool IsValidLanguageName(string? languageName)
+	public void Initialize(IncrementalGeneratorInitializationContext context)
 	{
-		try
-		{
-			if (languageName.IsNullOrEmpty())
-				return false;
-
-			if (languageName.StartsWith("qps-", StringComparison.Ordinal))
-				return true;
-
-			var culture = new CultureInfo(languageName);
-
-			while (!culture.IsNeutralCulture)
-				culture = culture.Parent;
-
-			return culture.LCID != 4096;
-		}
-		catch
-		{
-			return false;
-		}
-	}
-
-	// Code from: https://github.com/dotnet/ResXResourceManager/blob/0ec11bae232151400a5a8ca7b9835ac063c516d0/src/ResXManager.Model/ProjectFileExtensions.cs#L77
-	private static string GetBaseName(string filePath)
-	{
-		var name = Path.GetFileNameWithoutExtension(filePath);
-		var innerExtension = Path.GetExtension(name);
-		var languageName = innerExtension.TrimStart('.');
-
-		return IsValidLanguageName(languageName) ? Path.GetFileNameWithoutExtension(name) : name;
-	}
-
-	// Code from: https://github.com/dotnet/ResXResourceManager/blob/c8b5798d760f202a1842a74191e6010c6e8bbbc0/src/ResXManager.VSIX/Visuals/MoveToResourceViewModel.cs#L120
-	private static string GetLocalNamespace(string? resxPath, string? targetPath, string? projectPath, string? rootNamespace)
-	{
-		try
-		{
-			if (resxPath is null)
-				return string.Empty;
+		var globalOptions = context.AnalyzerConfigOptionsProvider.Select(GlobalOptions.Select).WithComparer(EqualityComparer<GlobalOptions>.Default);
 
-			var resxFolder = Path.GetDirectoryName(resxPath);
-			var projectFolder = Path.GetDirectoryName(projectPath);
-			rootNamespace ??= string.Empty;
+		var monitor = context.AdditionalTextsProvider
+			.Where(af => af.Path.EndsWith(".resx") && Path.GetFileNameWithoutExtension(af.Path) == Utilities.GetBaseName(af.Path));
 
-			if (resxFolder is null || projectFolder is null)
-				return string.Empty;
+		var inputs = monitor
+			.Combine(globalOptions)
+			.Combine(context.AnalyzerConfigOptionsProvider)
+			.Select((x,t)=>FileOptions.Select(x.Left.Left, x.Right, x.Left.Right, t))
+			.Where(x=>x.Valid);
```

Comment:
Style nit: Please place spaces before and after the `=>` operator.

Reply:
done